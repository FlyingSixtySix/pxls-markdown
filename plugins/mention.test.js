const { processor, expectProcessorParse } = require('../testUtil.js');
const mention = require('./mention.js');

processor.use(mention);

test('tokenizes "@mention" correctly', () => expectProcessorParse('@mention').toMatchSnapshot());

test('tokenizes "@Mock_Username-09" and calls mentionCallback with argument "Mock_Username-09"', () => {
	const mentionCallback = jest.fn();
	const processorWithCallback = processor().use(mention, { mentionCallback });
	processorWithCallback.parse('@Mock_Username-09');
	expect(mentionCallback).toHaveBeenNthCalledWith(1, 'Mock_Username-09');
});
