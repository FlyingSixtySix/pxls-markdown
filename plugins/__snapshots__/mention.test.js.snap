// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizes "@mention" correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "mention",
        "username": "mention",
        "value": "@mention",
      },
    ],
    "type": "paragraph",
  },
]
`;
